src/components/victory-bar/helper-methods.js

```diff
@@ -107,7 +104,12 @@ module.exports = {
       const allX = datasets.map((dataset) => {
```

Comment:
you could remove the flatten two lines below if you did
```js
datasets.reduce((array, dataset) => {
  return array.concat(dataset.data.map((d) => d.x));
}, [])
```
which should be more efficient (avoids iterating over the array to flatten it)

or this whole block could be made much more efficient (O(n) instead of O(n^2)) by using a Set (either the ES6 native set or a really small implementation of a Set with O(1) lookup)
(which considering this function is called from https://github.com/FormidableLabs/victory-chart/blob/b8eb4cad0feca11bdb51eb94d006921cc6cf9925/src/components/victory-bar/victory-bar.jsx#L285 matters a lot, because it'd make that renderBars() O(n^2) instead of O(n^3))